lightning/src/chain/chainmonitor.rs

```diff
@@ -170,6 +171,7 @@ where
 
 	fn block_disconnected(&self, header: &BlockHeader, height: u32) {
 		let monitors = self.monitors.read().unwrap();
+		log_debug!(self.logger, "Latest block disconnected, new best block {} at height {} provided via block_disconnected", header.block_hash(), height);
```

Comment:
I believe this should be the header and height of the disconnected block, not the new best block. We decrement the height and get the previous block hash in `ChannelMonitor::block_disconnected`.